### DISCUSSION TITLE
custom Sidebar

### DISCUSSION THREAD

**Discussion by ibtissemoussaifi** [stephanrauh/ngx-extended-pdf-viewer#1492] [Q&A]
Hello
Can you help me please ? :) 
I need pagination in sidebar
For example, in the first time i get only 25 pages in the sidebar, then when I click a button i get the other 25 pages.
How can i do it ?

Best Regards.

**Comment by stephanrauh** [comment]
In general, the thumbnails have decent performance, so I wonder what your use-case is. Even this document of 13234 pages works surprisingly well - it's a bit laggy, but it's still usable: https://github.com/stephanrauh/ngx-extended-pdf-viewer/issues/995 

Basically, you have to overwrite parts of the PDF_ThumbnailViewer class. JavaScript is flexible enough to allow you to do so. This `for` loop generates the list of thumbnails, so it's what you're looking for:

https://github.com/stephanrauh/pdf.js/blob/3bb60346573d760aa7f34875e51d31afa133dd90/web/pdf_thumbnail_viewer.js#L223
